### DISCUSSION TITLE
A way to keep bookmark toolbar from hiding with single toolbar setting on Linux

### DISCUSSION THREAD

**Discussion by istir** [zen-browser/desktop#3781] [Q&A]
Hello,
I'm using Zen on both a Mac and a Linux device. On a Mac, bookmark toolbar is always visible (when I set it to be "always visible" with Cmd+Shift+B. 
On Linux however, even after pressing that shortcut, toolbar is only visible after hovering on top of the browser. This happens even when bookmark toolbar is hidden altogether - I get this long-ish animation that shifts entire page down just to show me a close button. 

From what I've seen - this is caused by `#zen-appcontent-navbar-container` requiring `zen-has-hover="true"` attribute. Maybe there is a way for me to somehow always have this attribute added, without having to use multiple toolbar mode? 
I'm running `1.0.2-b.2 (Firefox 133.0.3) (64-bit)` on Hyprland. Not sure about other WMs.  

https://github.com/user-attachments/assets/97959ed8-f949-499a-8757-63b62f5556e8

**Comment by thunderhead07** [comment]
This also is the case on windows, it really bugs me.
I also like to see the close and minimise button all the time, but still want the Searchbar/Url box to be at the side.
It would be fantastic if that would be possible.

**Reply by istir** [reply]
Sadly shortly after after writing that it doesn't happen on macOS, I got an update and it started doing the same thing. Super jarring, so i stopped using single toolbar mode.

**Comment by dave-cometrics** [comment]
Any update on whether it's possible to do this without having to fork the whole project to change one tiny thing? This browser is _almost_ perfect, but there are all these little hyper-opinionated things about it that do not have associated settings that are keeping it from reaching that status.

**Comment by pssbg** [comment]
Also very much looking for this! Arc can't do it and this is perfect, just a shame that it seems to be buggy and hides instead of staying always visible.

**Comment by Zynera** [comment]
I think I found a way to fix this.

`zen.view.hide-window-controls false`

It doesn't let you use the compact mode and idk if it will work for linux but it works on windows
